### PR TITLE
[Profiler] A couple of error handling fixes

### PR DESCRIPTION
**Pull request by hamishknight**
Fix the handling of `try?` and throwing expressions in ternaries and if statement conditions.

rdar://118883320

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please build toolchain macOS

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please build toolchain macOS

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

**Review by bnbarham — APPROVED** [review]

File: test/Profiler/coverage_errors.swift [hunk 1]
```diff
@@ -400,6 +450,371 @@ func test23() throws -> Int { // CHECK-NEXT: [[@LINE]]:29 -> [[@LINE+11]]:2 : 0
...
+    ? throwingFn()            // CHECK-NEXT: [[@LINE]]:19   -> [[@LINE+2]]:11 : (0 - 2)
+    : throwingFn()            // CHECK-NEXT: [[@LINE]]:7    -> [[@LINE]]:19   : (0 - 1)
+  return x                    // CHECK-NEXT: [[@LINE-1]]:19 -> [[@LINE]]:11   : ((0 - 2) - 3)
+}                             // CHECK-NEXT: }
+
+// CHECK-LABEL: sil_coverage_map {{.*}} "$s15coverage_errors6test29Si_SityKF"
+func test29() throws -> (Int, Int) { // CHECK-NEXT: [[@LINE]]:36   -> [[@LINE+7]]:2  : 0
+  let x = try .random()              // CHECK-NEXT: [[@LINE+1]]:7  -> [[@LINE+1]]:24 : 1
+    ? (throwingFn(), 0)              // CHECK-NEXT: [[@LINE]]:20   -> [[@LINE]]:24   : (1 - 2)
+    : (0, throwingFn())              // CHECK-NEXT: [[@LINE-1]]:24 -> [[@LINE+1]]:11 : (0 - 2)
```
[the remark this answers is not in the document]
Comment (bnbarham) [reply test/Profiler/coverage_errors.swift:500]: I'm okay with `)`, I wrote the comment before seeing it and then added in that part after :)

**Pull request merged**